packages/experimental/src/vertical-css-transition/test/vertical-css-transition.tsx

```diff
@@ -34,7 +34,7 @@ describe( 'VerticalCSSTransition', () => {
 		}
 	} );
 
-	it( 'should set maxHeight of children to container on entering and remove it when entered', ( done ) => {
+	it.skip( 'should set maxHeight of children to container on entering and remove it when entered', ( done ) => {
```

Comment:
@louwie17 with an update to Jest 27, this test and the one below it fail with an error message suggesting the `expect` never gets executed.

```
thrown: "Exceeded timeout of 5000 ms for a test.
```

I skipped these so that I could verify everything else passes, but I want to address it ultimately. Is there way to explicitly fire the `entering` and `entered` states?

Reply:
Just a cosmetic idea, as we may touch this test:
WDYT of replacing `expect( count ).toEqual( 1 );` with slight more descriptive and stable `expect( onEntering ).toHaveBeenCalledTimes( 1 )` and add `expect( onEntered ).toHaveBeenCalledTimes( 1 )` ath the end of `it` to make sure both assertions are fired?

Reply:
That makes sense @tomalec, I will update that.

@psealock I can have a look at this, the `entering` and `entered` state should be triggered when the `in` property is toggle to `true`. As you can see we do this in the `rerender` function.

Reply:
@psealock I pushed up the fix for the failing tests. It was probably related to the same changes that required you to do this:

> This required removal of empty timeouts in favor of waitFor in order to clear queues of Promises.

The `entering` and `entered` states are triggered by `setTimeouts`, so calling `jest.runOnlyPendingTimers();` did the trick. I also updated the `count` variable name as Tomalec suggested.

Reply:
Awesome, thanks team! Changes look good to me. This is ready for wider review